official/keras_application_models/dataset.py

```diff
@@ -46,3 +47,31 @@ def generate_synthetic_input_dataset(model, batch_size):
       label_shape=tf.TensorShape(label_shape),
   )
   return dataset
+
+
+def generate_cifar10_dataset(batch_size):
+  """Generates CIFAR10 dataset.
+
+  Each sample consists of a 32x32 color image, and label is from 10 classes.
+
+  Args:
+    batch_size: int, the number of batch size.
+
+  Returns:
+    A tuple (train_dataset, test_dataset, image_shape, num_classes) where
+      train_dataset & test_dataset have shape ((batch, 32, 32, 3), (batch, 10)),
+      input_shape is the input image shape,
+      num_classes is number of output classes.
+  """
+  input_shape, num_classes = (32, 32, 3), 10
```

Comment:
These should be constants, outside of this function

Reply:
These two constants are tied to CIFAR-10 only, and has nothing to do with synthesis dataset aforementioned. I change all of these to instance variables. How does that look to you?